server/controllers/request.js

```diff
@@ -0,0 +1,87 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /requests?userid={}
+// @desc Search for request of especific userId
+// @access Private
+
+exports.getRequests = asyncHandler(async (req, res, next) => {
+    const userId = req.query.userid;
+    
+    let requests;
+    if (userId) {
+      requests = await Request.find({
+        userId: { $regex: userId, $options: "i" }
+      });
+    }
+  
+    if (requests.length==0) {
+      res.status(404);
+      throw new Error("No requests found for this user Id");
+    }
+  
+    res.status(200).json({ requests: requests });
+  });
+
+// @route POST /requests/new
+// @desc Post a new request 
+// @access Private
+
+exports.postRequest = asyncHandler(async (req, res, next) => {
+    const { userId, sitterId, start, end, status, paid } = req.body;
+    if (end<start){
+        res.status(400);
+          throw new Error("Invalid request dates");
+    };
+    const newRequest = await Request.create({
+        userId,
+        sitterId,
+        start,
+        end,
+        status,
+        paid
+      });
+   
+
+    if (newRequest) {
+        res.status(201).json({
+            success: {
+              newRequest: {
+                id: newRequest._id,
+              }
+            }
+          });
+        } else {
+          res.status(400);
+          throw new Error("Invalid request data");
+        }
```

Comment:
This check isn't necessary considering the way [how `await` works in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description) - if the call for `Request.create()` fails, the promise will reject and `await` will throw an error.

If the code successfully gets to `line 46`, you can assume the `Request` was created successfully. Otherwise, this handler will throw an error and `asyncHandler` will automatically forward this to the next handler (which in our case is the error handler).